Assets/Scripts/UI/SettingsMenu.cs

```diff
@@ -0,0 +1,145 @@
+            int resolution = PlayerPrefs.GetInt("Resolution", GetDefaultResolution());
+            
+            SetVolume(volume);
+            SetQuality(quality);
+            SetFullscreen(fullscreen);
+            SetResolution(resolution);
+            
+        }
+        #endregion
+
+        #region Settings
+                public void ApplySettings()
+        {
+            customSettings = true;
+            PlayerPrefs.SetInt("CustomSettings", customSettings ? 1 : 0);
+        }
+
+        private void SetHighestQuality()
+        {
+            int highestQuality = QualitySettings.GetQualityLevel();
+            QualitySettings.SetQualityLevel(highestQuality);
+            graphicsDropdown.value = highestQuality;
+        }
+
+        /// <summary>
+        /// Sets the volume of the game at a certain volume.
+        /// </summary>
+        /// <param name="volume">This is the volume's level.</param>
+        public void SetVolume(float volume)
+        {
+            audioMixer.SetFloat("volume", volume);
+            slider.value = volume;
+            PlayerPrefs.SetFloat("Volume", volume);
+        }
+
+        /// <summary>
+        /// Sets the quality of the game
+        /// </summary>
+        /// <param name="qualityIndex">This is the index of the quality found in project settings > Quality.</param>
+        public void SetQuality(int qualityIndex)
+        {
+            QualitySettings.SetQualityLevel(qualityIndex);
+            graphicsDropdown.value = qualityIndex;
+            PlayerPrefs.SetInt("Quality", qualityIndex);
+        }
+
+        /// <summary>
+        /// This allows the user to switch between fullscreen or not.
+        /// </summary>
+        /// <param name="isFullscreen">This bool checks if fullscreen is set or not</param>
+        public void SetFullscreen(bool isFullscreen)
+        {
+            Screen.fullScreen = isFullscreen;
+            PlayerPrefs.SetInt("Fullscreen", isFullscreen ? 1 : 0);
+        }
+
+        /// <summary>
+        /// This sets the resolution of the game.
+        /// </summary>
+        /// <param name="resolutionIndex">This index checks which resolution from the list it'll pick.</param>
+        public void SetResolution(int resolutionIndex)
+        {
+            Resolution resolution = _resolution[resolutionIndex];
+            Screen.SetResolution(resolution.width, resolution.height, Screen.fullScreen);
+            resolutionsDropdown.value = resolutionIndex;
+            PlayerPrefs.SetInt("Resolution", resolutionIndex);
+        }
+
+        private int GetDefaultResolution()
+        {
+            for (int i = 0; i < _resolution.Length; i++)
+            {
+                if (_resolution[i].width == Screen.currentResolution.width && 
+                    _resolution[i].height == Screen.currentResolution.height)
+                {
+                    return i;
+                }
+            }
```

Comment:
```suggestion
            int length = _resolution.Length; // zou ook "l" genoemd kunnen worden
            for (int i = 0; i < length; i++)
            {
                if (_resolution[i].width == Screen.currentResolution.width
                    && _resolution[i].height == Screen.currentResolution.height)
                {
                    return i;
                }
            }
```
[code convention](https://github.com/Team-Swamp/Project-VooDoo/wiki/Code-convention#loops)